test(e2e): run e2e tests on local env

alinasuciu opened the pull request:
Add the possibility to run e2e tests on local environment  
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-231402

cipak commented on webpack.config.js:
```diff
@@ -63,7 +63,6 @@ module.exports = function(env, argv) {
       argv.mode === 'development'
         ? {
             contentBase: path.resolve(__dirname, './demo'),
-            open: true,
```
does this mean that running `npm start` won't automatically open the demo page in the browser anymore?

cipak replied:
To keep this, we can use the `--open` cli option of webpack serve https://github.com/webpack/webpack-cli/blob/master/SERVE-OPTIONS.md

```json
  "scripts": {
    "demo:serve": "webpack-dev-server demo/App.jsx --mode development",
    "demo:build": "webpack demo/App.jsx --mode production",
    "start": "npm run demo:serve -- --open",
    "test:e2e": "cross-env START_SERVER_AND_TEST_INSECURE=1 start-server-and-test demo:serve https-get://localhost:9000 test:e2e",
    "test:eslint": "eslint src/"
  },
```

cipak replied:
And please leave `open: false` so that the intention not to open a browser is explicit.

lalli-flores reviewed (commented):
Not sure I'm thrilled with this solution. `cross-env` is actually not maintained anymore, so maybe we can just include the env variable in the env files?
I need to look a little into `start-server-and-test` to determine if it's a dependency we want to add. have you tried the static server from webdriver.io?

cipak commented on wdio.conf.js:
```diff
@@ -120,7 +120,15 @@ exports.config = {
+          { mount: '/', path: './docs' },
+        ]
+      },
+      'selenium-standalone'
```
i think `selenium-standalone` is a separate service and needs to be outside of `static-server`'s array

cipak commented on wdio.conf.js:
```diff
@@ -120,7 +120,15 @@ exports.config = {
+  services: [
+    ['static-server', {
+      folders: [
+          { mount: '/', path: './docs' },
```
indentation

alinasuciu commented:
> Not sure I'm thrilled with this solution. `cross-env` is actually not maintained anymore, so maybe we can just include the env variable in the env files?
> I need to look a little into `start-server-and-test` to determine if it's a dependency we want to add. have you tried the static server from webdriver.io?
I changed the implementation and now we are using static server from webdriverio. Thanks for the hint, I saw that they have this feature but I wasn't sure it would be what we need since we don't have static files. So in order for this to work and the tests to relay on the latest updates it's important to run the build before or I can also add the build command before the test command inside the script.

lalli-flores commented:
@alinasuciu , awesome! Glad it worked easily. A couple things:
* your commit should be of [`test` type](https://github.com/webex/components/blob/master/CONTRIBUTING.md#type)
* if building is necessary, then please do add the building script before starting the tests
* I'm ok with adding `docs` to `.gitignore`, but please do so in a separate commit. Also make sure to update the CI script for master updates to force add the docs folders since now it will be ignored

alinasuciu commented:
> @alinasuciu , awesome! Glad it worked easily. A couple things:
> 
> * your commit should be of [`test` type](https://github.com/webex/components/blob/master/CONTRIBUTING.md#type)
> * if building is necessary, then please do add the building script before starting the tests
> * I'm ok with adding `docs` to `.gitignore`, but please do so in a separate commit. Also make sure to update the CI script for master updates to force add the docs folders since now it will be ignored


@lalli-flores , I added the build script before the test and changed the commit label. Regarding the docs file in gitignore, I'll create a new task for it and a new pr. It's important to have this one merged.

webex-components-publisher commented:
:tada: This PR is included in version 1.4.1 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@webex/widgets/v/1.4.1)
- [GitHub release](https://github.com/webex/widgets/releases/tag/v1.4.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
